False positive in empty_enum_arguments rule

Unrelated to #1549.

file:

```swift
import Foundation

let myString = "FOO"

switch myString {
case "foo".uppercased(): // ← this triggers
    print("shouty")
default:
    break
}
```

```sh
bash-3.2$ swiftlint version
0.19.0
bash-3.2$ swiftlint lint --path empty_enum_arguments.swift 
Linting Swift files at path empty_enum_arguments.swift
Linting 'empty_enum_arguments.swift' (1/1)
/Users/zev/Desktop/empty_enum_arguments.swift:6:22: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
Done linting! Found 1 violation, 0 serious in 1 file.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -29,7 +29,9 @@
 
 ##### Bug Fixes
 
-* None.
+* Fix false positive in `empty_enum_arguments` rule when calling methods.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#1597](https://github.com/realm/SwiftLint/issues/1597)
 
 ## 0.19.0: Coin-Operated Machine
 

--- a/Source/SwiftLintFramework/Rules/EmptyEnumArgumentsRule.swift
+++ b/Source/SwiftLintFramework/Rules/EmptyEnumArgumentsRule.swift
@@ -22,7 +22,8 @@ public struct EmptyEnumArgumentsRule: ASTRule, ConfigurationProviderRule, Correc
             "switch foo {\n case .bar: break\n}",
             "switch foo {\n case .bar(let x): break\n}",
             "switch foo {\n case let .bar(x): break\n}",
-            "switch (foo, bar) {\n case (_, _): break\n}"
+            "switch (foo, bar) {\n case (_, _): break\n}",
+            "switch foo {\n case \"bar\".uppercased(): break\n}"
         ],
         triggeringExamples: [
             "switch foo {\n case .bar↓(_): break\n}",
@@ -59,6 +60,17 @@ public struct EmptyEnumArgumentsRule: ASTRule, ConfigurationProviderRule, Correc
 
         let contents = file.contents.bridge()
 
+        let callsRanges = dictionary.substructure.flatMap { dict -> NSRange? in
+            guard dict.kind.flatMap(SwiftExpressionKind.init) == .call,
+                let offset = dict.offset,
+                let length = dict.length,
+                let range = contents.byteRangeToNSRange(start: offset, length: length) else {
+                    return nil
+            }
+
+            return range
+        }
+
         return dictionary.elements.flatMap { subDictionary -> [NSRange] in
             guard subDictionary.kind == "source.lang.swift.structure.elem.pattern",
                 let offset = subDictionary.offset,
@@ -77,6 +89,10 @@ public struct EmptyEnumArgumentsRule: ASTRule, ConfigurationProviderRule, Correc
                 // avoid matches after `where` keyworkd
                 if let whereMatch = file.match(pattern: "where", with: [.keyword], range: caseRange).first,
                     whereMatch.location < range.location {
+                    return nil
+                }
+
+                if callsRanges.first(where: { range.intersects($0) }) != nil {
                     return nil
                 }
 

